SCloudRandomizer/MainViewController.m

```diff
@@ -310,86 +309,98 @@ - (void)getTrackInfo:(NSDictionary *)track shouldPlay:(BOOL)play
     NSString *streamURL = [track objectForKey:@"stream_url"];
     NSLog(@"The streamURL is: %@", streamURL);
     
+    // Get new track if stream URL is nil
     while ([GlobalData stringIsNilOrEmpty:streamURL])
     {
         NSLog(@"streamURL is null");
-        [self refreshTrackList];
-        NSInteger randomNumber = arc4random_uniform((uint32_t) self.tracks.count);
-        self.currentSongNumber = randomNumber;
-        NSDictionary *track2 = [self.tracks objectAtIndex:randomNumber];
+        [self getTracks:NO shouldPlay:NO];
+        NSDictionary *track2 = [self.tracks objectAtIndex:self.currentSongNumber];
         streamURL = [track2 objectForKey:@"stream_url"];
     }
     
+    // Progress/loading hud
     SCAccount *account = [SCSoundCloud account];
     MBProgressHUD *hud = [MBProgressHUD showHUDAddedTo:self.view animated:YES];
     hud.mode = MBProgressHUDModeIndeterminate;
     hud.labelText = @"Loading Track";
     hud.detailsLabelText = @"Please wait..";
     
+    // Progress handler
```

Comment:
We can move out the progress handler logic into its own class and have the main view controller use it when it needs. We can do this in a separate PR